ports/renesas-ra/modpyb.c

```diff
@@ -0,0 +1,261 @@
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(pyb_fault_debug_obj, pyb_fault_debug);
+
+#if MICROPY_PY_PYB_LEGACY
+
+// Returns the number of milliseconds which have elapsed since `start`.
+// This function takes care of counter wrap and always returns a positive number.
+STATIC mp_obj_t pyb_elapsed_millis(mp_obj_t start) {
+    uint32_t startMillis = mp_obj_get_int(start);
+    uint32_t currMillis = mp_hal_ticks_ms();
+    return MP_OBJ_NEW_SMALL_INT((currMillis - startMillis) & 0x3fffffff);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(pyb_elapsed_millis_obj, pyb_elapsed_millis);
+
+// Returns the number of microseconds which have elapsed since `start`.
+// This function takes care of counter wrap and always returns a positive number.
+STATIC mp_obj_t pyb_elapsed_micros(mp_obj_t start) {
+    uint32_t startMicros = mp_obj_get_int(start);
+    uint32_t currMicros = mp_hal_ticks_us();
+    return MP_OBJ_NEW_SMALL_INT((currMicros - startMicros) & 0x3fffffff);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(pyb_elapsed_micros_obj, pyb_elapsed_micros);
+
+#endif
+
+MP_DECLARE_CONST_FUN_OBJ_KW(pyb_main_obj); // defined in main.c
+
+// Get or set the UART object that the REPL is repeated on.
+// This is a legacy function, use of uos.dupterm is preferred.
+STATIC mp_obj_t pyb_repl_uart(size_t n_args, const mp_obj_t *args) {
+    if (n_args == 0) {
+        if (MP_STATE_PORT(pyb_stdio_uart) == NULL) {
+            return mp_const_none;
+        } else {
+            return MP_OBJ_FROM_PTR(MP_STATE_PORT(pyb_stdio_uart));
+        }
+    } else {
+        if (args[0] == mp_const_none) {
+            if (MP_STATE_PORT(pyb_stdio_uart) != NULL) {
+                uart_attach_to_repl(MP_STATE_PORT(pyb_stdio_uart), false);
+                MP_STATE_PORT(pyb_stdio_uart) = NULL;
+            }
+        } else if (mp_obj_get_type(args[0]) == &pyb_uart_type) {
+            MP_STATE_PORT(pyb_stdio_uart) = MP_OBJ_TO_PTR(args[0]);
+            uart_attach_to_repl(MP_STATE_PORT(pyb_stdio_uart), true);
+        } else {
+            mp_raise_ValueError(MP_ERROR_TEXT("need a UART object"));
+        }
+        return mp_const_none;
+    }
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(pyb_repl_uart_obj, 0, 1, pyb_repl_uart);
+
+STATIC mp_obj_t pyb_country(size_t n_args, const mp_obj_t *args) {
+    if (n_args == 0) {
+        return mp_obj_new_str(pyb_country_code, 2);
+    } else {
+        size_t len;
+        const char *str = mp_obj_str_get_data(args[0], &len);
+        if (len != 2) {
+            mp_raise_ValueError(NULL);
+        }
+        pyb_country_code[0] = str[0];
+        pyb_country_code[1] = str[1];
+        return mp_const_none;
+    }
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(pyb_country_obj, 0, 1, pyb_country);
+
+STATIC const mp_rom_map_elem_t pyb_module_globals_table[] = {
+    { MP_ROM_QSTR(MP_QSTR___name__), MP_ROM_QSTR(MP_QSTR_pyb) },
+    { MP_ROM_QSTR(MP_QSTR_fault_debug), MP_ROM_PTR(&pyb_fault_debug_obj) },
+    #if MICROPY_PY_PYB_LEGACY
```

Comment:
All of these legacy parts of this `pyb` module should be removed, they are deprecated and there are other ways to access these methods (eg `machine.bootloader()`, `machine.freq()`).

Reply:
OK, I have remoted them.